kernel/src/snapshot.rs

```diff
@@ -215,18 +215,25 @@ impl Snapshot {
             new_protocol,
             new_log_segment.end_version,
         )?;
+
         // NB: we must add the new log segment to the existing snapshot's log segment
         let mut ascending_commit_files = old_log_segment.ascending_commit_files.clone();
         ascending_commit_files.extend(new_log_segment.ascending_commit_files);
         let mut ascending_compaction_files = old_log_segment.ascending_compaction_files.clone();
         ascending_compaction_files.extend(new_log_segment.ascending_compaction_files);
+
+        let latest_crc_file = new_log_segment
```

Comment:
should we ensure that we don't go backwards here by mistake?

Reply:
ah nice, thanks guys I didn't really consider this until now! agree it sounds like the right thing is to just use the new (older) CRC in case the new segment is missing a later CRC?

and yup that's correct: all our log listing boils down to `list_log_files` which will filter based on end version. I think that's also a good call-out but generally we have enough semantics baked in to `LogSegment` and listing that I don't think we really every create a 'bad state' which could have a CRC somehow included after the end version

Reply:
(btw added a comment to document this)